internal/monitors/vsphere/monitor.go

```diff
@@ -0,0 +1,38 @@
+package vsphere
+
+import (
+	"context"
+	"time"
+
+	"github.com/signalfx/signalfx-agent/internal/monitors/vsphere/model"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+)
+
+type Monitor struct {
+	Output types.Output
+	cancel func()
+}
+
+func init() {
+	monitors.Register(
+		&monitorMetadata,
+		func() interface{} { return &Monitor{} },
+		&model.Config{},
+	)
+}
+
+func (m *Monitor) Configure(conf *model.Config) error {
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+	r := newRunner(ctx, conf, m)
+	utils.RunOnInterval(ctx, r.run, time.Duration(conf.IntervalSeconds)*time.Second)
```

Comment:
This will default to 10 seconds. Is that going to cause problems with the 20 second metric cycle in vSphere?

Reply:
Should be fine. As above, if it's not time to fetch data, then it won't try.